### DISCUSSION TITLE
GPU power usage much lower in Edge main profile vs. other Edge profile and other browsers

### DISCUSSION THREAD

**Discussion by Weissrolf** [foundryvtt/foundryvtt#11694] [Q&A] [answered]
Ahoi. I stumbled over an odd observation and haven't been able yet to find the reasons. I already dug down to browser extensions, browser settings and Foundry core settings plus trying tabbed vs. dedicated browser windows, but didn't find any explanation yet.

The following graphs show GPU power usage while running a FoundryVTT client session with various mods. All variants display the very same view (zoom and panning), except for fog effects (stacked Foundry + FX module) adding some random element.

Edge main user profile (MS cloud profile) LEFT vs. Edge new local profile CENTER vs. Edge main profile RIGHT:
![image](https://github.com/user-attachments/assets/f1ce798e-2fb7-42f7-99aa-288ede82e36e)

Firefox:
![image](https://github.com/user-attachments/assets/ea55b35a-037d-4803-a7df-2530d1867977)

Brave:
![image](https://github.com/user-attachments/assets/1dd79904-f941-4f1e-b15f-2f75dc888fa5)

Chrome (RIGHT part of the graph vs. Edge main profile on the left):
![image](https://github.com/user-attachments/assets/2ac13beb-e592-4a4b-8f17-0f97ae7f7d42)

Any idea how to get all browser variants to use as little GPU power (load) as my main Edge profile?

**Comment by Weissrolf** [comment]
Moved to issues since discussions don't seem to be used here (no answers for any discussion topics).

**Comment by aaclayton** [comment]
We can use a discussion since it's unclear there is an issue here. Can you please double check that Foundry is using the same GPU in both profiles? If you have two GPUs available (for example one discrete and one integrated) it is possible that different browser performance profiles could prefer one vs the other.

Otherwise, the performance profile used would be partially determined based on the `powerPreference` setting passed in to the `PIXI.Application` constructor. See <https://api.pixijs.io/@pixi/app/PIXI/Application.html>

In the Foundry codebase this is defined in `Canvas.#configureCanvasSettings()`. We set `powerPreference: "high-performance"` but this is just a recommendation/request for PIXI to honor that. It's possible that certain browser profiles would restrict GPU resources. 

Try experimenting with editing the `foundry.js` file to set some different values for `powerPreference` as documented here: https://pixijs.download/v7.x/docs/PIXI.IRendererOptions.html. Report back if you observe a difference.

**Comment by Weissrolf** [comment]
Hi Andrew,

thanks for the quick reply.

This was tested on my Desktop PC that uses a 4090 with disabled (!) Intel iGPU. And the power readings are the ones reported by the 4090 specifically. So it's always the same GPU.

At first I thought it should be browser settings, because the new local Edge profile resets all settings to defaults, but that's not it. Then I thought it would be Foundry core settings (like fps, quality, midmap) that may be saved per browser session even for the same client/user, but - unless I overlooked something - these don't seem to be it either (I ran all clients at 60 fps and High quality for the tests). Next I thought that this might be a "dedicated browser windows vs. tab" thing, but when I moved the "main profile" Edge tab out to its own window it still used the lower GPU load vs. all other variants.

I will try to follow your instructions concerning the powerPreference thingy and report back.

**Comment by Weissrolf** [comment]
It's noteworthy that the much lower powered "main profile" Edge browser still maintains the 60 fps. I also did a run without modules (via FindTheCulprit) and the results are the same. Both the GM and player sessions run below 20 W for my main Edge profile, but over 40 W + spikes for the local profile.

![image](https://github.com/user-attachments/assets/be349ea5-4f5f-4e86-a683-52b5729278db) ![image](https://github.com/user-attachments/assets/2710c672-3600-45d7-9cab-896886c6bdf1)

Take note of the lower "memory controller" load in the low powered variant! On average the difference is less, but still something like <20% vs. >35%.  I don't know if the memory controller also changes frequency, something I can test, though.

**Reply by Weissrolf** [reply]
I set powerPreference in foundry.js (had to search for it) to "low-power". Still using 2 times the power in the local profile window vs. my main profile, including higher memory controller load. So it made no difference. Same goes for Chrome.

**Comment by Weissrolf** [comment]
I disabled GPU power-saving (fixed base GPU frequency) via Nvidia driver to get more comparable numbers:

![image](https://github.com/user-attachments/assets/3f9c5051-5a21-4131-bab0-9a589247178d) ![image](https://github.com/user-attachments/assets/a3ff87bf-126e-4922-bb21-941fd89e9178)

Results for the 2nd Edge profile and Chrome are the same (higher results).

**Comment by Weissrolf** [comment, marked as answer]
Since this had to be a "per client" thing that I overlooked instead of "per Edge profile" one, I deleted the Foundry client cookie. And indeed my main Edge profile's GPU usage increased!

Looking at things again I noticed that the "Per Pixel Resolution Ratio" was not disabled in the other browsers/profiles. I will do more comparisons to check if this was the only difference I overlooked (I know that I properly looked at Quality and FPS limiting).

**Comment by Weissrolf** [comment]
Yep, looks like user error based on the "Per Pixel Resolution Ratio" setting, despite me knowing about it and supposedly looking at all performance based core settings. Sorry for the confusion and thanks for the help!

PS: I will keep the powerPreference hint in, because I usually run all of this on an underpowered laptop (trying to find a new one capable of running 2x 2K screens). I assume this will be overwritten with the next update, though, and with the new laptop I might prefer the dGPU anyway (14th gen Intel iGPUs still struggle with Foundry at 3x 2K).
